src/lint/rules/variable-use-def.ts

```diff
@@ -0,0 +1,352 @@
+      );
+      if (paramsIndex !== -1) {
+        for (
+          ;
+          paramsIndex < parts.length &&
+          !(
+            parts[paramsIndex].name === 'text' &&
+            (parts[paramsIndex].contents as string).includes(')')
+          );
+          ++paramsIndex
+        ) {
+          const v = parts[paramsIndex];
+          if (isVariable(v)) {
+            acScope.declare(v.contents[0].contents, v, 'abstract closure parameter');
+          }
+        }
+      }
+      let capturesIndex = parts.findIndex(
+        p => p.name === 'text' && p.contents.endsWith(' that captures ')
+      );
+
+      if (capturesIndex !== -1) {
+        for (; capturesIndex < parts.length; ++capturesIndex) {
+          const v = parts[capturesIndex];
+          if (v.name === 'text' && v.contents.includes(' and performs ')) {
+            break;
+          }
+          if (isVariable(v)) {
+            const name = v.contents[0].contents;
+            scope.use(v);
+            acScope.declare(name, v, 'abstract closure capture');
+          }
+        }
+      }
+
+      // we have a lint rule elsewhere which checks there are substeps for closures, but we can't guarantee that rule hasn't tripped this run, so we still need to guard
+      if (step.sublist != null && step.sublist.name === 'ol') {
+        walkAlgorithm(step.sublist, acScope, report);
+        for (const [name, { node, kind, used }] of acScope.vars) {
+          if (kind === 'abstract closure capture' && !used) {
+            report({
+              ruleId: 'unused-capture',
+              message: `closure captures ${JSON.stringify(name)}, but never uses it`,
+              line: node!.location.start.line,
+              column: node!.location.start.column,
+            });
+          }
+        }
+      }
+      continue;
+    }
+
+    // handle let/such that/there exists declarations
+    for (let i = 1; i < parts.length; ++i) {
+      const part = parts[i];
+      if (isVariable(part) && !loopVars.has(part)) {
+        const varName = part.contents[0].contents;
+
+        // check for "there exists"
+        const prev = parts[i - 1];
+        if (
+          prev.name === 'text' &&
+          /\b(?:for any |there exists |there is |there does not exist )(\w+ )*$/.test(prev.contents)
+        ) {
+          scope.declare(varName, part);
+          declaredThisLine.add(part);
+          continue;
+        }
+
+        // check for "Let _x_ be" / "_x_ and _y_ such that"
+        if (i < parts.length - 1) {
+          const next = parts[i + 1];
+          const isSuchThat = next.name === 'text' && next.contents.startsWith(' such that ');
+          const isBe = next.name === 'text' && next.contents.startsWith(' be ');
+
+          if (isSuchThat || isBe) {
+            const varsDeclaredHere = [part];
+            let varIndex = i - 1;
+            // walk backwards collecting this comma/'and' seperated list of variables
```

Comment:
I'd kind of rather get rid of these. We can leave this support for now, but can you open an issue on 262 that lists these cases so we can review them for phrasing?

Reply:
Not sure about an issue on 262, but here's a list for you.

- [This one line in JSON.stringify](https://github.com/tc39/ecma262/pull/2900) which should just get fixed
- Module Record [GetBindingValue](https://tc39.es/ecma262/multipage/executable-code-and-execution-contexts.html#sec-module-environment-records-getbindingvalue-n-s) step 3.a
- [Number::toString](https://tc39.es/ecma262/multipage/ecmascript-data-types-and-values.html#sec-numeric-types-number-tostring) step 5
- [Number.prototype.toExponential](https://tc39.es/ecma262/multipage/numbers-and-dates.html#sec-number.prototype.toexponential) step 10
- [Number.prototype.toPrecision](https://tc39.es/ecma262/#sec-number.prototype.toprecision) step 10

In Temporal there is also [ParseTimeZoneOffsetString](https://tc39.es/proposal-temporal/#sec-temporal-parsetimezoneoffsetstring) step 3.

Reply:
(Note that  [PR #2288](https://github.com/tc39/ecma262/pull/2288) takes care of Module Record GetBindingValue step 3.a.)